src/hit/api/params.cpp

```diff
@@ -42,22 +42,34 @@ namespace hit {
         return x - repr(x, n) + 1;
     }
 
-    uint64_t nextPrime(uint64_t last_prime, uint64_t two_n, const vector<Modulus> &mods) {
-        // get the closest prime to `last_prime` that is also congruent to 1 mod 2N
-        uint64_t next_test_val = to1Coset(last_prime);
+    // Find the smallest prime larger than `x` that is not in `mods`.
+    uint64_t nextPrime(uint64_t x, uint64_t n, const vector<Modulus> &mods) {
+        // find a number near x that is congruent to 1 mod n
```

Comment:
Algorithm 3 from the paper looks for the first prime modulus such that it is congruent to 1 mod 2N. This changes that guarantee. How do we know this is safe?

Or have we not actually changed this because we previously were accidentally finding answers congruent to 1 mod 4N and we're just fixing it?